File: 0148-sort-list/0148-sort-list.java
class Solution {
    public ListNode sortList(ListNode head) {
        List<Integer> ans = new ArrayList<>();
        
        ListNode curr = head;
        while (curr != null) {
            ans.add(curr.val);
            curr = curr.next;
        }
        
        Collections.sort(ans);
        
        curr = head;
        int i = 0;
        while (curr != null) {
            curr.val = ans.get(i++);
            curr = curr.next;
        }
        
        return head;
    }
}

File: 0148-sort-list/0148-sort-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        ListNode* temp = head; 
        vector<int>ans; 

        while(temp!= NULL){
            ans.push_back(temp->val); 
            temp = temp->next; 
        }
        sort(ans.begin(),ans.end()); 
        temp = head; 
        int i = 0; 
        while(temp!= NULL){
            temp->val = ans[i++]; 
            temp = temp->next; 
        }
        return head; 
        
    }
};